### PR TITLE
DRUP-242: Updated context api client: added method for updating resource state

### PR DESCRIPTION
**Pull request by PavelLoparev**

### CODE REVIEW

File: src/Context/ContextApi.php [hunk 1]
```diff
@@ -296,4 +297,20 @@ public function renderContext($contextUid)
         return $this->sendRequest($endpoint, $requestData, self::HTTP_METHOD_POST);
     }
 
+    /**
+     * Update resource state.
+     *
+     * @param $resourceId
```
Comment (vsolovei-smartling) [inline src/Context/ContextApi.php]:
```suggestion
     * @param string $resourceId
```

File: examples/context-example.php [hunk 2]
```diff
@@ -274,6 +275,37 @@ function uploadResourceDemo($authProvider, $projectId, $resourceId, $fileUri) {
     return $response;
 }
 
+/**
+ * @param \Smartling\AuthApi\AuthApiInterface $authProvider
+ * @param string $projectId
+ * @param string $resourceId
+ * @return bool
+ */
+function updateResourceStateDemo($authProvider, $projectId, $resourceId) {
```
Comment (vsolovei-smartling) [inline examples/context-example.php]: Please specify type here as well
```suggestion
function updateResourceStateDemo(\Smartling\AuthApi\AuthApiInterface $authProvider, $projectId, $resourceId) {
```

File: src/Context/ContextApi.php [hunk 3]
```diff
@@ -296,4 +297,20 @@ public function renderContext($contextUid)
...
+     * @param UpdateResourceStateParameters $params
+     * @return bool
+     * @throws SmartlingApiException
+     */
+    public function updateResourceState($resourceId, UpdateResourceStateParameters $params)
+    {
+        $endpoint = \vsprintf('resources/%s/state', $resourceId);
```
Comment (vsolovei-smartling) [inline src/Context/ContextApi.php]: Should be one of
```
        $endpoint = \vsprintf('resources/%s/state', [$resourceId]);
```
(vsprintf expects second argument to be array, will fail with "vsprintf(): Argument #2 ($values) must be of type array, string given" if strict types are enabled or with later php versions)
```
        $endpoint = \sprintf('resources/%s/state', $resourceId);
```
or
```suggestion
        $endpoint = "resources/$resourceId/state";
```

**Review by vsolovei-smartling — CHANGES_REQUESTED** [review]

File: src/Context/ContextApi.php (hunk 1 above)
Comment (PavelLoparev) [reply src/Context/ContextApi.php]: Done

File: examples/context-example.php (hunk 2 above)
Comment (PavelLoparev) [reply examples/context-example.php]: Done

File: src/Context/ContextApi.php (hunk 3 above)
Comment (PavelLoparev) [reply src/Context/ContextApi.php]: Thanks, done

**Review by vsolovei-smartling — APPROVED** [review]

**Pull request merged**